crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -438,7 +438,7 @@ fn constrain<'a, P>(expr_parser: P) -> impl NoirParser<Statement> + 'a
 where
     P: ExprParser + 'a,
 {
-    ignore_then_commit(keyword(Keyword::Constrain).labelled("statement"), expr_parser)
+    ignore_then_commit(choice((keyword(Keyword::Assert), keyword(Keyword::Constrain))).labelled("statement"), expr_parser)
```

Comment:
For future note, if there are only two items you can also use `keyword(Keyword::Assert).or(keyword(Keyword::Constrain)`